[ONNX] Fix the warnings of `aten overload fallback to default` in onnx dispatcher

titaiwangms opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #105972

Without this PR, the warning message is misleading as it says the default is found before the error message popped.
Next PR will start refactoring aten overload fallback with adding overloads supported by torchlib into OpSchema matching.

titaiwangms committed:
[ONNX] Fix atethe warnings of a overload fallback to default in onnxdispatcher

[ghstack-poisoned]

BowenBao reviewed (approved):
LG, please rephrase title xd.

BowenBao commented on torch/onnx/_internal/fx/onnxfunction_dispatcher.py:
```diff
@@ -310,19 +310,19 @@ def get_function_overloads(
+                diagnostic = diagnostic_context.inflight_diagnostic()
+                diagnostic.with_additional_message(
+                    "### The operator overload is not found in onnx registry!\n"
+                    "Cannot find the operator overload in onnx registry, but"
```
```suggestion
                    "Cannot find the operator overload in onnx registry, but "
```

titaiwangms committed:
Update on "[ONNX] Fix the warnings of `aten overload fallback to default` in onnx dispatcher"


Without this PR, the warning message is misleading as it says the default is found before the error message popped.
Next PR will start refactoring aten overload fallback with adding overloads supported by torchlib into OpSchema matching.

[ghstack-poisoned]

titaiwangms commented:
@pytorchbot merge

titaiwangms commented:
@pytorchbot merge
